React/wbooks/src/BookDetail/index.js

```diff
@@ -1,55 +1,57 @@
 import React, { Component } from 'react';
-import { Link, withRouter } from 'react-router-dom';
+import { Link } from 'react-router-dom';
 import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 
 import { BOOKS } from '../books_json';
-import { bookDetailAction } from '../redux/bookDetail/actions';
+import ActionCreators from '../redux/books/actions';
 
 import BookDetailInfo from './Components/BookDetailInfo';
 import BookSuggestions from './Components/BookSuggestions';
 import CommentSection from './Components/CommentSection';
 
 class BookDetail extends Component {
-  componentWillMount() {
-    this.props.dispatch(bookDetailAction.bookDetail(this.props.match.params.id));
+  componentDidMount() {
+    this.props.dispatch(ActionCreators.bookDetail(this.props.match.params.id));
   }
   render() {
-    const bookId = this.props.match.params.id;
-    const book = this.props.items && this.props.items.data ? this.props.items.data : null;
-    const suggestions = book ? (
-      <BookSuggestions suggestions={BOOKS.filter(b => b.genre == book.genre)} />
-    ) : null;
-
-    const bookComponent = book ? (
-      <BookDetailInfo
-        imageUrl={book.image_url}
-        title={book.title}
-        author={book.author}
-        year={book.year}
-        genre={book.genre}
-      />
-    ) : null;
+    const book = this.props.items ? this.props.items : null;
 
     return (
       <div>
         <div className="body-container">
           <Link to="/" className="back-button">
             &lt; Volver
           </Link>
-          {bookComponent}
-          <hr />
-          {suggestions}
+          {book && (
+            <div>
+              <BookDetailInfo
+                imageUrl={book.image_url}
+                title={book.title}
+                author={book.author}
+                year={book.year}
+                genre={book.genre}
+              />
+              <hr />
+              <BookSuggestions suggestions={BOOKS.filter(b => b.genre === book.genre)} />
+            </div>
+          )}
           <hr />
           <CommentSection />
         </div>
       </div>
     );
   }
 }
+
+BookDetail.propTypes = {
+  items: PropTypes.arrayOf
```

Comment:
Be as specific as possible with the propTypes. 

```
items: PropTypes.arrayOf(PropTypes.shape({ image_url: PropTypes.string, title: PropTypes.string .... }))
```

You can even have a propTypes folder in src that has a file called Book.js with book prop types. That'd be a nice to have to avoid repeating the proptypes